src/components/Quotes.js

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react';
+
+function Quotes() {
+  const [quoteData, setQuoteData] = useState(['Quote here']);
+  const [error, setError] = useState(0);
+  const [loading, setLoading] = useState(0);
+
+  useEffect(() => {
+    setLoading(1);
+    fetch('https://api.api-ninjas.com/v1/quotes?category=intelligence', {
+      headers: {
+        'X-Api-Key': 'l7joY6pxRAtosVpy+GGSVg==tL66pxRY0ioTzgkd',
+      },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          setError(1);
+        }
+        return res.json();
+      })
+      .then((data) => {
+        if (Array.isArray(data) && data.length > 0) {
+          setLoading(0);
+          setQuoteData(data[0]);
+        } else {
+          setError(1);
+        }
+      })
+      .catch((error) => {
+        setError(error);
+      });
+  }, []);
```

Comment:
[OPTIONAL] This is optional but recommended so please don't skip this requirement. Well done :clap: you did a very nice job :+1: but I recommended using `async` and `await` because the purpose of `async / await` is to simplify the syntax necessary to consume promise-based APIs. The behaviour of `async / await` is similar to combining generators and promises. `Async` functions always return a promise. Also, create a function and call that function in the `useEffect` then your code looks clean :+1: check below :point_down:  my suggestion :slightly_smiling_face: 

```suggestion
  useEffect(() => {
    useEffect(() => {
  const fetchData = async () => {
    try {
      setLoading(1);
      const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=intelligence', {
        headers: {
          'X-Api-Key': 'l7joY6pxRAtosVpy+GGSVg==tL66pxRY0ioTzgkd',
        },
      });
      
      if (!response.ok) {
        throw new Error('Response not OK');
      }

      const data = await response.json();

      if (Array.isArray(data) && data.length > 0) {
        setLoading(0);
        setQuoteData(data[0]);
      } else {
        throw new Error('Empty data');
      }
    } catch (error) {
      setError(error);
    }
  };

  fetchData();
}, []);
```